webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { startCase } from 'lodash';
+import { sprintf, translate as __ } from 'foremanReact/common/I18n';
+import { Button } from 'patternfly-react';
+import ForemanModal from 'foremanReact/components/ForemanModal';
+import { useForemanModal } from 'foremanReact/components/ForemanModal/ForemanModalHooks';
+import { getURIQuery } from 'foremanReact/common/helpers';
+import {
+  CANCEL_MODAL,
+  RESUME_MODAL,
+  CANCEL_SELECTED_MODAL,
+  RESUME_SELECTED_MODAL,
+} from '../../TasksTableConstants';
+
+export const ConfirmModal = ({
+  actionText,
+  actionState,
+  actionType,
+  selectedRowsLen,
+  id,
+  bulkCancelById,
+  bulkCancelBySearch,
+  bulkResumeById,
+  bulkResumeBySearch,
+  cancelTask,
+  resumeTask,
+  parentTaskID,
+  allRowsSelected,
+  getSelectedTasks,
+  clicked,
+  url,
+}) => {
+  const uriQuery = getURIQuery(url);
+  const tasksActions = {
+    [CANCEL_SELECTED_MODAL]: () =>
+      allRowsSelected
+        ? bulkCancelBySearch({ query: uriQuery, parentTaskID })
+        : bulkCancelById({
+            selected: getSelectedTasks(),
+            url,
+            parentTaskID,
+          }),
+    [CANCEL_MODAL]: () =>
+      cancelTask({
+        taskId: clicked.taskId,
```

Comment:
can you move this action into a `ConfirmModalActions` file,
for example it can be implemented like this: https://pastebin.com/3zR6j9qW
and than import it in the `index.js` file

Reply:
Moved, thanks!